### DISCUSSION TITLE
rustic v0.9.2 / v0.9.3

### DISCUSSION THREAD

**Discussion by simonsan** [rustic-rs/rustic#1310] [Announcements]
**UPDATE**: rustic v0.9.2 introduced an issue that was breaking everyone's workflow. This has been fixed now in a patch release: https://github.com/rustic-rs/rustic/releases/tag/v0.9.3

Hello `rustic`eans,  
this release might look small, but it contains a very exciting new feature: 🪝Hooks.

Hooks are commands that are run during certain events in the application lifecycle. They can be used to run custom scripts or commands before or after certain actions.

The documentation you can find in our [user docs](https://rustic.cli.rs/docs/commands/misc/hooks.html) and in the [config docs](https://github.com/rustic-rs/rustic/tree/main/config#global-hooks-globalhooks).

@mueckinger also contributed a CI workflow to automatically build and publish a [Docker image](https://github.com/rustic-rs/rustic/pkgs/container/rustic) on release! So you can easily use rustic now in your Docker setup. Thank you for that!

We hope this release is helpful to you, if you are experiencing any issues, please open an issue in our [issue tracker](https://github.com/rustic-rs/rustic/issues/new)!

# Changelog

### Features

- *(config)* Add hooks ([#1218](https://github.com/rustic-rs/rustic/pull/1218))

### Miscellaneous

- *(deps)* update rustic_core ([#1309](https://github.com/rustic-rs/rustic/pull/1309))
- build and publish docker image on release ([#1297](https://github.com/rustic-rs/rustic/pull/1297))

<hr /><em>This discussion was created from the release <a href='https://github.com/rustic-rs/rustic/releases/tag/v0.9.2'>rustic v0.9.2</a>.</em>

**Comment by aawsome** [comment]
Here are some use cases which might be interesting to use hooks for:

Global hooks:
- Send messages after successful / failed runs
- Feed your custom logging/monitoring with information about start/end of rustic calls

Repository hooks:
- Mount/umount the drive where the repository is located
- Sync your repo to some remote destination after each command
- Start extra integrity checks like checking the SHA256 against repo files 

Backup hooks:
- Mount/umount a drive with data-to-backup
- Run commands to save some systems state, e.g. `dpkg --get-selections` on debian-based systems
- Dump a database into a file and remove the dump after backup (using `stdin-command` may be an alternative)

If you have a nice use-case yourself, please share it with others!

**Reply by simonsan** [reply]
I added your comment to the documentation: https://rustic.cli.rs/docs/commands/misc/hooks.html#use-cases

**Reply by nickchomey** [reply]
How can we use the repository hooks to create then umount and remove a directory used for `rustic mount`  

I just did
mkdir ~/mnt
rustic mount ~/mnt

then closed the mount and the mnt directory still exists. I need to `umount ~/mnt && rm -rf ~/mnt` but would like to have this happen automatically

I would only want this to happen when the mount is closed
